File: hsc-tracker-v2/update_auth.js
const fs = require('fs');
let code = fs.readFileSync('src/contexts/AuthContext.tsx', 'utf-8');
code = code.replace(
  'signIn: () => Promise<void>;', 
  'signIn: () => Promise<User | null>;'
);
code = code.replace(
  '  const signIn = async () => {\n    try {\n      await signInWithPopup(auth, googleProvider);\n    } catch (error) {\n      console.error("Error signing in", error);\n    }\n  };',
  '  const signIn = async () => {\n    try {\n      const cred = await signInWithPopup(auth, googleProvider);\n      return cred.user;\n    } catch (error) {\n      console.error("Error signing in", error);\n      return null;\n    }\n  };'
);
fs.writeFileSync('src/contexts/AuthContext.tsx', code);

File: hsc-tracker-v2/update_auth.py
import re

with open('src/contexts/AuthContext.tsx', 'r') as f:
    code = f.read()

code = code.replace(
    'signIn: () => Promise<void>;', 
    'signIn: () => Promise<User | null>;'
)

old_fn = """  const signIn = async () => {
    try {
      await signInWithPopup(auth, googleProvider);
    } catch (error) {
      console.error("Error signing in", error);
    }
  };"""

new_fn = """  const signIn = async () => {
    try {
      const cred = await signInWithPopup(auth, googleProvider);
      return cred.user;
    } catch (error) {
      console.error("Error signing in", error);
      return null;
    }
  };"""

code = code.replace(old_fn, new_fn)

with open('src/contexts/AuthContext.tsx', 'w') as f:
    f.write(code)

